sdks/go/test/integration/primitives/windowinto.go

```diff
@@ -91,3 +91,35 @@ func WindowSums_GBK(s beam.Scope) {
 func WindowSums_Lifted(s beam.Scope) {
 	WindowSums(s.Scope("Lifted"), stats.SumPerKey)
 }
+
+// TriggerWindowSums, much like WindowSums described above has an addition of configuring
+// a trigger here. SetDefault works fine. Other triggers such as SetAlways throws
+// pane decoding error.
+func TriggerWindowSums(s beam.Scope, sumPerKey func(beam.Scope, beam.PCollection) beam.PCollection) {
+	timestampedData := beam.ParDo(s, &createTimestampedData{Data: []int{4, 9, 2, 3, 5, 7, 8, 1, 6}}, beam.Impulse(s))
+
+	windowSize := 3 * time.Second
+
+	validate := func(s beam.Scope, wfn *window.Fn, in beam.PCollection, expected ...interface{}) {
+		// Window the data.
+		windowed := beam.WindowInto(s, wfn, in)
+
+		// change below statement to: windowed.WindowingStrategy().SetAlways()
+		// to get the decoding error.
+		windowed.WindowingStrategy().SetDefault()
```

Comment:
OK. It pains me to suggest this, but this user experience leaves a little to be desired and we have a short window before the Go SDK exits experimental where we can make it better. 

I propose we make a `beam.WindowIntoOption` interface type, and change `beam.WindowInto` to take a variadic parameter of those to configure the trigger and other windowing strategy properties. Then we can type assert and apply the options to the windowing strategy without much concern.

This would be similar to how `beam.ParDo` has an option type https://github.com/apache/beam/blob/master/sdks/go/pkg/beam/option.go#L25
https://github.com/apache/beam/blob/master/sdks/go/pkg/beam/pardo.go#L356

My recommendation would be to make that into a separate change by itself, not defining any of the concrete `beam.WindowIntoOption` types and handling, and just changing the signature to take in `...options beam.WindowIntoOption` as the final parameter (variadic parameters must be the final one in a function signature).

While this change would not be 100% backwards compatible, it would be source compatible for all reasonable uses of `beam.WindowInto`. The main place it would break is if a user were doing something like:

`var myWindowInto func(s Scope, ws *window.Fn, col PCollection) PCollection = beam.WindowInto`

which is unlikely and weird to begin with (it wouldn't serve anything beyond ensuring the signature never changes).

Then we can rebase and get back to here and finish handling the implementation.